docs/UserGuide.md

```diff
@@ -72,295 +82,468 @@ Information that is important to know to prevent unexpected or undefined behavio
 
 Format: `delete c INDEX`
-* `INDEX` refers to the index number shown in the currently displayed contact list of companies.
-* Deletes the contact of the company at the specified `INDEX` in the currently displayed contact list of companies.
 
-Parameter
-* `INDEX` is compulsory.
-    * `INDEX` must be a positive integer and cannot be larger than the size of the currently displayed contact list of companies.
+<div markdown="span" class="alert alert-info">
+**:pencil: Parameter Information**<br>
+* The `INDEX` must be:
+    * a positive integer (e.g. 0, 1, 2…) with a maximum value of 2147483647
+    * smaller than the size of the list of companies currently displayed
+* The `INDEX` depends on the currently displayed list of companies which changes as
+you sort companies or search for specific companies
+</div><br>
 
 Example
-* `list c` followed by `delete c 2` deletes the second company in the list:
+* `list c` followed by `delete c 2`: Deletes the second company in the list of companies
   ![delete company at index 2](images/deleteCompany.png)
 
+[Back to Table of Contents](#table-of-contents)
+
 ### Deleting an internship from company: `delete i`
 
-Keep your list of internships for companies organised by deleting internships you no longer need.
+Keep your list of internships for each company organised by deleting internships you are no longer interested in.
 
-Format: `delete i c/INDEX i/INDEX`
-* `c/INDEX` refers to the index number shown in the currently displayed contact list of companies.
-* `i/INDEX` refers to the index number shown in the currently displayed internship list of the target company.
-* Deletes the internship with index `INDEX` of the company at the specified `INDEX` in the currently displayed contact list of companies.
+Format: `delete i c/COMPANY_INDEX i/INTERNSHIP_INDEX`
 
-Parameter
-* The two `INDEX` fields are both compulsory.
-    * `INDEX` with `c/` prefix must be a positive integer and cannot be larger than the size of the currently displayed contact list of companies.
-    * `INDEX` with `i/` prefix must be a positive integer and cannot be larger than the size of the currently displayed internship list of the target company.
+<div markdown="span" class="alert alert-info">
+**:pencil: Parameter Information**<br>
+* The `COMPANY_INDEX` and `INTERNSHIP_INDEX` must each be:
+    * a positive integer (e.g. 0, 1, 2…) with a maximum value of 2147483647
+    * smaller than the size of their respective lists currently displayed 
+* The `COMPANY_INDEX` depends on the currently displayed list of companies which changes as
+you sort companies or search for specific companies
+</div><br>
 
 Example
-* `list c` and `view c 2` followed by `delete i c/2 i/1` deletes the first internship of the second company in the list:
+* `list c` followed by `delete i c/2 i/1`: Deletes the first internship of the second company in the list of companies
```

Comment:
Other than my earlier point about moving the list commands to be before delete, do you think it is helpful to put a heading + description for each command type? e.g.

### List commands
View all your contacts at a glance

* `list p`: etc.
* `list c `: etc.